app/controllers/users.js

```diff
@@ -1,47 +1,105 @@
 import Ember from 'ember';
 import DS from 'ember-data';
 
-const {computed, inject, RSVP} = Ember;
-const {service} = inject;
-const {PromiseArray} = DS;
+const { computed, Controller, inject, isEmpty, RSVP, run } = Ember;
+const { Promise } = RSVP;
+const { service } = inject;
+const { PromiseArray } = DS;
+const { debounce } = run;
+const { trim } = Ember.$;
+
+function cleanQuery(query) {
+  return trim(query).replace(/[\-,?~!@#$%&*+\-'="]/, '');
+}
+
+export default Controller.extend({
+  init() {
+    this._super(...arguments);
+
+    // This needs to be removed after user session re-write
+    this.get('currentUser.model').then((user) => {
```

Comment:
Its not a good temporary solution either.  Having a promise in init follows under the same logic as observers in general.  It makes side effects very difficult to understand.  Any promises needed before the controller is build needs to go into the route.